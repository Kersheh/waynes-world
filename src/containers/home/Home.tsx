import React from 'react';
import { useSelector } from 'react-redux';

import { RootState } from 'types';
import AlbumTile from './albumTile/AlbumTile';
import styles from './Home.module.scss';

const HomeContainer = () => {
  const { recentlyAddedAlbums } = useSelector(
    (state: RootState) => state.library
  );

  return (
    <div className={styles.home}>
      <h2>Recently added</h2>

      {recentlyAddedAlbums.length === 0 && (
        <p>No recently added albums found.</p>
      )}

      <div className={styles.albumScroll}>
        {recentlyAddedAlbums?.map(
          (album, index) =>
            !(index % 2) && (
              <AlbumTile
                album={album.album}
                artist={album.artist}
                id={album.id}
                key={`${album.album}-${index}`}
              />
            )
        )}
      </div>

      <div className={styles.albumScroll}>
        {recentlyAddedAlbums?.map(
          (album, index) =>
            !!(index % 2) && (
              <AlbumTile
                album={album.album}
                artist={album.artist}
                id={album.id}
                key={`${album.album}-${index}`}
              />
            )
        )}
      </div>
    </div>
  );
};

export default HomeContainer;
