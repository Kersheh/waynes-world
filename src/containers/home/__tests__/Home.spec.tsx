import React from 'react';
import { render } from '@testing-library/react';

import Home from '../Home';

describe('<Home/> component', () => {
  it('should render', () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });
});
