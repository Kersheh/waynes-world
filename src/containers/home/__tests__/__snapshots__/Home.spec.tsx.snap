// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> component should render 1`] = `
<div>
  <div
    class="home"
  >
    <h2>
      Recently added
    </h2>
    <p>
      No recently added albums found.
    </p>
    <div
      class="albumScroll"
    />
    <div
      class="albumScroll"
    />
  </div>
</div>
`;
